### PR TITLE
Removing Webhooks and Adding SCM

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Build Webhook was removed in https://github.com/screwdriver-cd/screwdriver/pull/153

Adding SCM for https://github.com/screwdriver-cd/scm-base/pull/5

### CODE REVIEW

File: core/scm.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        .example('https://github.com/stjohnjohnson'),
+
+    name: Joi.string()
+        .required()
+        .label('Display Name')
+        .example('stjohnjohnson'),
+
+    avatar: Joi.string()
+        .uri()
+        .required()
```
Comment (d2lam) [inline core/scm.js]: This seems specific for github. Do we know other source control would have an "avatar" for users?

File: core/scm.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        .label('Link to Repository')
+        .example('https://github.com/screwdriver-cd/screwdriver/tree/master')
+}).label('SCM Repository');
+
+const SCHEMA_COMMIT = Joi.object().keys({
+    sha: Joi.string()
+        .hex()
+        .length(40)
+        .required()
+        .label('SHA1 identifier for the commit')
```
Comment (d2lam) [inline core/scm.js]: `SHA` instead of `SHA1`

Comment (stjohnjohnson) [inline core/scm.js]: But it is a SHA1, anything else won't be 40 characters.

File: core/scm.js (hunk 1 above)
Comment (stjohnjohnson) [inline core/scm.js]: Pretty much all of them: 
 - https://docs.gitlab.com/ee/customization/libravatar.html
 - https://blog.bitbucket.org/2013/07/16/customize-your-avatar-in-bitbucket-no-gravatar-needed/

**Comment by d2lam** [comment]
Yeah I'm fine with this 👍

**Comment by FenrirUnbound** [comment]
👍 

Looks good to me

this is fine l - imgur

**Pull request merged**